app/models/appointment.rb

```diff
@@ -148,7 +145,8 @@ def anonymized_data
   end
 
   def previously_communicated_via?(communication_type)
-    communications.latest_by_type(communication_type)&.attempted?
+    matching_notifications = notifications.includes(:communications).where(communications: {communication_type: communication_type})
+    matching_notifications.any? { |n| n.communication.attempted? }
```

Comment:
I don't think this quite matches the previous logic ?

Reply:
I think it achieves the same thing. Unrelated, but I think "attempted" is a poorly named method. It's defined as 
```
  def attempted?
    successful? || in_progress?
  end
  ```
  which means failures weren't attempted. So it's really more like "not failed".